airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/JdbcSqlOperations.java

```diff
@@ -148,16 +149,20 @@ public boolean isValidData(final JsonNode data) {
 
   @Override
   public final void insertRecords(final JdbcDatabase database,
-                                  final List<AirbyteRecordMessage> records,
+                                  final List<PartialAirbyteMessage> records,
                                   final String schemaName,
                                   final String tableName)
       throws Exception {
-    dataAdapter.ifPresent(adapter -> records.forEach(airbyteRecordMessage -> adapter.adapt(airbyteRecordMessage.getData())));
+    dataAdapter.ifPresent(adapter -> records.forEach(airbyteRecordMessage -> {
```

Comment:
the only DataAdapter in existence is destination-postgres, because of https://airbytehq-team.slack.com/archives/C03C4AVJWG4/p1700605909300539 / https://github.com/airbytehq/airbyte/issues/3476

so I don't feel bad about de+reserializing here